k9mail/src/main/java/com/fsck/k9/K9.java

```diff
@@ -465,6 +469,9 @@ public static void save(StorageEditor editor) {
         editor.putBoolean("hideUserAgent", mHideUserAgent);
         editor.putBoolean("hideTimeZone", mHideTimeZone);
 
+        editor.putString("cryptoProvider", sCryptoProvider);
```

Comment:
What's to say the provider can't also provide other crypto? So far I put OpenPGP only in places where actual openpgp things happened, and use "crypto" where it's mostly generic. See `CryptoError` and such. Right now I'd say it's more likely that OpenKeychain grows S/MIME support than someone coming along with an api of their own for this.